avm/res/azure-stack-hci/marketplace-gallery-image/main.bicep

```
@@ -0,0 +1,173 @@
+        '/providers/Microsoft.Authorization/roleDefinitions/'
+      )
+      ? roleAssignment.roleDefinitionIdOrName
+      : subscriptionResourceId('Microsoft.Authorization/roleDefinitions', roleAssignment.roleDefinitionIdOrName))
+  })
+]
+
+// ============== //
+// Resources      //
+// ============== //
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.azurestackhci-markplgalleryimg.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource marketplaceGalleryImages 'Microsoft.AzureStackHCI/marketplaceGalleryImages@2024-10-01-preview' = {
+  extendedLocation: {
+    name: customLocationResourceId
+    type: 'CustomLocation'
+  }
+  location: location
+  name: name
+  properties: marketplaceGalleryImageProperties
+  tags: tags
+}
+
+resource marketplaceGalleryImages_roleAssignments 'Microsoft.Authorization/roleAssignments@2022-04-01' = [
+  for (roleAssignment, index) in (formattedRoleAssignments ?? []): {
+    name: roleAssignment.?name ?? guid(
+      marketplaceGalleryImages.id,
+      roleAssignment.principalId,
+      roleAssignment.roleDefinitionId
+    )
+    properties: {
+      roleDefinitionId: roleAssignment.roleDefinitionId
+      principalId: roleAssignment.principalId
+      description: roleAssignment.?description
+      principalType: roleAssignment.?principalType
+      condition: roleAssignment.?condition
+      conditionVersion: !empty(roleAssignment.?condition) ? (roleAssignment.?conditionVersion ?? '2.0') : null // Must only be set if condtion is set
+      delegatedManagedIdentityResourceId: roleAssignment.?delegatedManagedIdentityResourceId
+    }
+    scope: marketplaceGalleryImages
+  }
+]
+
+// ============ //
+// Outputs      //
+// ============ //
+
+@description('The name of the marketplace gallery images.')
+output name string = marketplaceGalleryImages.name
+
+@description('The resource ID of the marketplace gallery images.')
+output resourceId string = marketplaceGalleryImages.id
+
+@description('The resource group of the marketplace gallery images.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The location of the marketplace gallery images.')
+output location string = marketplaceGalleryImages.location
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+type galleryOSDiskImageType = {
+  // Note: The API definition shows this as empty, so I'm defining it as an empty object
+}
```

If empty, it might have made sense to not surface it in general. But according to [this source](https://learn.microsoft.com/en-us/rest/api/stackhci/marketplace-gallery-images/create-or-update?view=rest-stackhci-2024-01-01&tabs=HTTP#galleryosdiskimage), there's actually something here:
```suggestion
@export()
@description('The type of the OS disk image.')
type galleryOSDiskImageType = {
   @description('Optional. The size of the VHD to be created.')
   sizeInMB: int?
}
```